tests/base/src/test/java/org/wildfly/security/auth/FileSystemSecurityRealmTest.java

```diff
@@ -158,35 +176,55 @@ public void testCreateAndLoadIdentityEncryption() throws Exception {
         securityRealm = FileSystemSecurityRealm.builder()
                 .setRoot(getRootPath(false))
                 .setLevels(3)
-                .setSecretKey(key)
                 .setProviders(ELYTRON_PASSWORD_PROVIDERS)
+                .setSecretKey(secretKey)
                 .build();
         ModifiableRealmIdentity existingIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
         assertTrue(existingIdentity.exists());
         existingIdentity.dispose();
     }
 
     @Test
-    public void testShortUsername() throws Exception {
-        FileSystemSecurityRealm securityRealm = new FileSystemSecurityRealm(getRootPath(), 3, ELYTRON_PASSWORD_PROVIDERS);
-        ModifiableRealmIdentity newIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("p"));
+    public void testCreateAndLoadIdentityIntegrity() throws Exception {
+        FileSystemSecurityRealm securityRealm = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath())
+                .setLevels(3)
+                .setPrivateKey(privateKey)
+                .setPublicKey(publicKey)
+                .setProviders(ELYTRON_PASSWORD_PROVIDERS)
+                .build();
+        ModifiableRealmIdentity newIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
         newIdentity.create();
-
         newIdentity.dispose();
 
-        ModifiableRealmIdentity existingIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("p"));
+        securityRealm = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath(false))
+                .setLevels(3)
+                .setPrivateKey(privateKey)
+                .setPublicKey(publicKey)
+                .setProviders(ELYTRON_PASSWORD_PROVIDERS)
+                .build();
```

Comment:
Just a general comment. Since the same builders are being used in multiple places, might be better to introduce methods that return the two common builders.

Reply:
Fixed